samza-core/src/main/scala/org/apache/samza/config/ShellCommandConfig.scala

```diff
@@ -50,6 +50,17 @@ object ShellCommandConfig {
     */
   val ENV_EXECUTION_ENV_CONTAINER_ID = "EXECUTION_ENV_CONTAINER_ID"
 
+  /**
+   * Set to "true" if application master isolation is enabled. Otherwise, will be considered false.
+   */
+  val ENV_APPLICATION_MASTER_ISOLATION_ENABLED = "APPLICATION_MASTER_ISOLATION_ENABLED"
+
+  /**
+   * Can be set to a directory which is different from the base lib directory which contains some additional
+   * application-specific resources.
```

Comment:
Can you clarify (in docs) what this env variable is used for? E.g., what's a 'base lib directory', when would you set it to a non-'base lib directory', and what additional application-specific resources? Also, why would you do this?